### PR TITLE
Update eini package

### LINKED ISSUE #307: eini hex.pm package causes any dependency on erlcloud package to break
Due to eini not depending completely on hex.pm packages, the usage of a package that depends on erlcloud's hex.pm package breaks with an error like:

    ==> eini (compile)
    Dependency not available: proper-.* ({git,
                                          "git://github.com/manopapad/proper.git",
                                          {tag,"1.1.1-beta"}})
    Dependency not available: covertool-.* ({git,
                                             "https://github.com/idubrov/covertool.git",
                                             {branch,"master"}})
    ERROR: compile failed while processing .../deps/eini: rebar_abort
    ** (Mix) Could not compile dependency :eini, "rebar compile skip_deps=true deps_dir=".../_build/dev/lib"" command failed. You can recompile this dependency with "mix deps.compile eini", update it with "mix deps.update eini" or clean it with "mix deps.clean eini"

### PR DESCRIPTION
**Pull request by waisbrot**
Update to use the erlcloud/eini repo and the 1.2.2 tag

Fixes #307

### CODE REVIEW

File: rebar.config.script [hunk 1]
```diff
@@ -8,7 +8,7 @@ case erlang:function_exported(rebar3, main, 1) of
         [{deps, [{meck, ".*",{git, "https://github.com/eproxus/meck.git", {tag, "0.8.4"}}},
                  {jsx, ".*", {git, "git://github.com/talentdeficit/jsx.git", {tag, "2.8.0"}}},
                  %% {hackney, ".*", {git, "git://github.com/benoitc/hackney.git", {tag, "1.2.0"}}},
-                 {eini, ".*", {git, "https://github.com/accense/eini.git", {tag, "1.2.1"}}},
+                 {eini, ".*", {git, "https://github.com/erlcloud/eini.git", {branch, "1.2.2"}}},
```
Comment (motobob) [inline rebar.config.script]: s/branch/tag since https://github.com/erlcloud/eini/releases/tag/1.2.2

**Comment by motobob** [comment]
👍  but let's fix tests. Strange travis tests failures. 
works just fine on my machine too...

**Comment by waisbrot** [comment]
Pretty weird! Yecc is inserting some `-dialyzer` lines to ignore the generated code, but Erlang 17 doesn't support that. I'm guessing that rebar3 itself is getting run under Erlang 18 and that's why we get the bad lines.

**Comment by motobob** [comment]
I have not yet found the root cause of this but given the fact eini is "ours" now I would consider removing that nonsense there as I do not want to lose value of having automated eunit done by travis for `erlcloud` PRs

**Comment by waisbrot** [comment]
I tried this locally, and couldn't reproduce it -- running rebar3 under Erlang 17.5 doesn't produce the `dialyzer` attributes. So I assume it's a bug in the script I wrote to get Travis to run it.

I can take another stab at tracking the problem down this this week, or we could just turn off the 17.5+rebar3 tests for now.

**Comment by waisbrot** [comment]
Tests pass with the updated `eini` package!

**Comment by motobob** [comment]
(yey) 👍

File: rebar.config.script [hunk 2]
```diff
@@ -8,7 +8,7 @@ case erlang:function_exported(rebar3, main, 1) of
         [{deps, [{meck, ".*",{git, "https://github.com/eproxus/meck.git", {tag, "0.8.4"}}},
                  {jsx, ".*", {git, "git://github.com/talentdeficit/jsx.git", {tag, "2.8.0"}}},
                  %% {hackney, ".*", {git, "git://github.com/benoitc/hackney.git", {tag, "1.2.0"}}},
-                 {eini, ".*", {git, "https://github.com/accense/eini.git", {tag, "1.2.1"}}},
+                 {eini, ".*", {git, "https://github.com/erlcloud/eini.git", {tag, "1.2.2"}}},
```
Comment (motobob) [inline rebar.config.script]: s/1.2.2/1.2.3 ?

Comment (motobob) [inline rebar.config.script]: @waisbrot last change here and it's ready to be merged

Comment (waisbrot) [inline rebar.config.script]: D'oh!  I need copy over that fancier `rebar.config.script` :-)

**Pull request merged**